solr/templates/storage.yaml

```diff
@@ -0,0 +1,30 @@
+{{ if (eq .Values.solr_mode "local") }}
```

Comment:
I don't think that it is the responsibility of the `solr` chart to provision the StorageClasses that it uses, it's down to the kubernetes admin to make sure that the Storage Classes are there with the name that they choose. We can provide documentation that makes recommendations about the configuration of the storage classes, but the solr statefulset should allow the `storageClassName` to be set.